Sources/client-sdk-swift/momento/TopicClient.swift

```diff
@@ -11,19 +15,52 @@ public class TopicClient: TopicClientProtocol {
     var credentialProvider: CredentialProviderProtocol
     
     @available(macOS 10.15, *)
-    init(configuration: TopicClientConfiguration, credentialProvider: CredentialProviderProtocol, logger: MomentoLoggerProtocol = DefaultMomentoLogger(loggerName: "swift-momento-logger", level: DefaultMomentoLoggerLevel.info)) {
+    init(
+        configuration: TopicClientConfiguration,
+        credentialProvider: CredentialProviderProtocol,
+        logger: MomentoLoggerProtocol = DefaultMomentoLogger(
+            loggerName: "swift-momento-logger",
+            level: DefaultMomentoLoggerLevel.info
+        )
+    ) {
         self.logger = logger
         self.credentialProvider = credentialProvider
-        self.pubsubClient = PubsubClient(logger: logger, configuration: configuration, credentialProvider: credentialProvider)
+        self.pubsubClient = PubsubClient(
+            logger: logger,
+            configuration: configuration,
+            credentialProvider: credentialProvider
+        )
     }
     
-    public func publish() async -> String {
+    public func publish(
+        cacheName: String,
+        topicName: String,
+        value: String
+    ) async -> TopicPublishResponse {
+        if cacheName.count < 1 {
+            return TopicPublishError(
+                error: SdkError(
```

Comment:
Whenever possible we should use the specific error subclasses in these cases (which is actually possible now that you marked them all public-- thanks!): `error: InvalidArgumentError("Must provide a cache name")`. That way the errorCode (and messageWrapper) will already be correctly configured for the error.